client/my-sites/comments/comment/comment-author.jsx

```diff
@@ -0,0 +1,111 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import Gridicon from 'gridicons';
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Emojify from 'components/emojify';
+import ExternalLink from 'components/external-link';
+import Gravatar from 'components/gravatar';
+import { convertDateToUserLocation } from 'components/post-schedule/utils';
+import { gmtOffset, timezone } from 'lib/site/utils';
+import { urlToDomainAndPath } from 'lib/url';
+import { getSiteComment } from 'state/selectors';
+import { getSite } from 'state/sites/selectors';
+import { getAuthorDisplayName, getGravatarUser, getPostTitle } from './utils';
+
+export const CommentAuthor = ( {
+	authorAvatarUrl,
+	authorDisplayName,
+	authorUrl,
+	commentDate,
+	commentType,
+	commentUrl,
+	isExpanded,
+	moment,
+	postTitle,
+	site,
+} ) => {
+	const gravatarUser = getGravatarUser( authorAvatarUrl, authorDisplayName );
+
+	const localizedDate = convertDateToUserLocation(
+		commentDate || moment(),
```

Comment:
What type is commentDate? A date instance, string, etc?

Reply:
It's a string.
Although, to be honest, I tried to simply copy-paste this kind of functions (especially those not authored by me originally—or those I don't remember authoring, like this 😄 ), to avoid breaking on some forgotten edge case.

In this particular case, I'm not sure why we'd need the `|| moment()`, unless `convertDateToUserLocation` might break on undefined date (which can happen while the comment loads).
Though, I'd say that here no date is better than "now".